lib/mempool/composedFees.js

```diff
@@ -0,0 +1,290 @@
+   * @param {Boolean} current - Whether the chain is synced.
+   */
+
+  processTX(entry, current) {
+    this.blockspaceFees.processTX(entry, current);
+    if (rules.countOpens(entry.tx) > 0)
+      this.openFees.processTX(entry, current);
+    if (rules.countUpdates(entry.tx) > 0)
+      this.updateFees.processTX(entry, current);
+    if (rules.countRenewals(entry.tx) > 0)
+      this.renewalFees.processTX(entry, current);
+  }
+
+  /**
+   * Process an entry being removed from the mempool.
+   * @param {Number} height - Block height.
+   * @param {MempoolEntry} entry
+   */
+
+  processBlockTX(height, entry) {
+    this.blockspaceFees.processBlockTX(height, entry);
+    if (rules.countOpens(entry.tx) > 0)
+      this.openFees.processBlockTX(height, entry);
+    if (rules.countUpdates(entry.tx) > 0)
+      this.updateFees.processBlockTX(height, entry);
+    if (rules.countRenewals(entry.tx) > 0)
+      this.renewalFees.processBlockTX(height, entry);
+  }
+
+  /**
+   * Process a block of transaction entries being removed from the mempool.
+   * @param {Number} height - Block height.
+   * @param {MempoolEntry[]} entries
+   * @param {Boolean} current - Whether the chain is synced.
+   */
+
+  processBlock(height, entries, current) {
+    const openEntries = [];
+    const updateEntries = [];
+    const renewalEntries = [];
+    for (const entry of entries) {
+      if (rules.countOpens(entry.tx) > 0)
+        openEntries.push(entry);
+      if (rules.countUpdates(entry.tx) > 0)
+        updateEntries.push(entry);
+      if (rules.countRenewals(entry.tx) > 0)
+        renewalEntries.push(entry);
+    }
+
+    this.blockspaceFees.processBlock(height, entries, current);
+    this.openFees.processBlock(height, openEntries, current);
+    this.updateFees.processBlock(height, updateEntries, current);
+    this.renewalFees.processBlock(height, renewalEntries, current);
+  }
+
+  /**
+   * Estimate a fee rate.
+   * @param {Number} [target=1] - Confirmation target.
+   * @param {Boolean} [smart=true] - Smart estimation.
+   * @param {Number?} - Covenant Type.
+   * @returns {Rate}
+   */
+
+  estimateFee(target, smart, type) {
+    if (!target)
+      target = 1;
+
+    if (smart == null)
+      smart = true;
+
+    const estimates = [];
+    estimates.push(this.blockspaceFees.estimateFee(target, smart));
+
+    if (rules.isOpenLimited(type))
+      estimates.push(this.openFees.estimateFee(target, smart));
+    if (rules.isUpdateLimited(type))
+      estimates.push(this.updateFees.estimateFee(target, smart));
+    if (rules.isRenewalLimited(type))
+      estimates.push(this.renewalFees.estimateFee(target, smart));
+
+    return Math.max(...estimates);
+  }
+
+  /**
+   * Estimate a priority.
+   * @param {Number} [target=1] - Confirmation target.
+   * @param {Boolean} [smart=true] - Smart estimation.
+   * @param {String?} - Covenant Type.
+   * @returns {Number}
+   */
+
+  estimatePriority(target, smart, covenantType) {
```

Comment:
I think `estimatePriority` is deprecated and removed in Bitcoin Core:

https://github.com/bitcoin/bitcoin/blob/452bb90c718da18a79bfad50ff9b7d1c8f1b4aa3/doc/release-notes/release-notes-0.15.0.md#removal-of-coin-age-priority

In bcoin and hsd, it's still exposed by the API but not used anywhere like `estimateFee` is...

So, maybe this is a bigger discussion, but we can probably forget this for now.